mitodl/open-discussions
# Update react query

#### Pre-Flight checklist


#### What are the relevant tickets?
#3981 

#### What's this PR do?
This PR updates react-query to v4. See code comments for specific changes.

#### How should this be manually tested?
There should be no change in behavior. I recommend testing...
- `/infinite/search`
- `/infinite/lists` ... try creating/editing a list. Updates should be visible.
- **Definitely try sorting items:**
    - Create a learning path at `/infinite/lists` if you don't have one already. Populate it with some items via the bookmark icon on search page / home page learning resource cards
    - View the details page for the learning path. _Details page navigable from `/infinite/lists`_
    - Click "Reorder" and try changing the item order. Re-ordering should be smooth. Items **should not snap back briefly** when released.
